# Exploring the Indian Startup Ecosystem: A Data Driven Analysis of Funding Trends and Industry Sectors

## Business Understanding
### Business Scenario
Your team is trying to venture into the Indian start-up
ecosystem. As the data expert of the team, you are to
investigate the ecosystem and propose the best course
of action.

*Analyze funding received by start-ups in India from
2018 to 2021.*
- Separate data for each year of funding will is
provided.
- In these datasets, you'll find the start-ups' details,
the funding amounts received, and the investors'
information.

### Business Objective
The aim of this project is to perform analysis on the Indian start-ups ecosystem and advice stakeholders on which venture to invest in to increase the potential of high profit/income.

## Hypothesis Testing
*Hypothesis* - The amount of funds a company receive depends on the sector a company finds itself
- Null Hypothesis(H_o) - The funds a company receive does not depend on the sector of investment
- Alternate Hypothesis(H_a) - The funds a company receive depends on the sector of investment

### Business Questions
- Which particular sector received the most funding over the time frame?
- The distribution of start ups in stages and the amount allocated each?
- What is the distribution of fundings based on locations?
- Which year had the most investors?
- Top 3 investor considerations in investing in start ups
- What was the impact of COVID-19 pandemic on startup funding in 2020?


# Data Loading and Exploration


## Data Dictionary

| Column Names|Description| Data Type|
|-------------|-----------|----------|
|Company/Brand|Name of company/start-up|Object|
|Founded|Year Start-up was founded|Int|
|Sector| Sector of service|Object|
|What it Does|Description about the company|Object|
|Founders| Founders of the company|Object|
|Investor| Investors|Object|
|Amount($)|Raised Founds|Float|
|Stage|Round of funding reached|Object|
|Headquarter|Location of company|Object|

## Data Understanding
Data Understanding phase drives the focus to identify, collect, and analyse
the data sets that can help accomplish the project goals. This
phase also has four tasks:


In [1]:
#import all necessary libraries

# data manipulation
import pandas as pd
import numpy as np

# data visualization libraries
import matplotlib.pyplot as plt
from plotly import express as px
import seaborn as sns

# statistical libraries
from scipy import stats
import statistics as stat

# database manipulation libraries
import pyodbc
from dotenv import dotenv_values

# hide warnings
import warnings
warnings.filterwarnings("ignore")
 

C:\Users\Alphagoal\AppData\Local\Temp\ipykernel_17260\2529016175.py:4: DeprecationWarning: 
Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),
(to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)
but was not found to be installed on your system.
If this would cause problems for you,
please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466
        
  import pandas as pd


## Setup Database Connection

In [2]:
# load environment variables
environment_variables = dotenv_values(".env")

# load database configurations
database = environment_variables.get("DB_DATABASENAME")
server = environment_variables.get("DB_SERVER")
username = environment_variables.get("DB_USERNAME")
password = environment_variables.get("DB_PASSWORD")

# database connection string
connection_string = f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password}"



In [3]:
# create pyodbc connector
connection = pyodbc.connect(connection_string)


In [4]:
# Loading 2021 dataset from MS SQL server
query_2021 = " SELECT * FROM dbo.LP1_startup_funding2021"
df_2021 = pd.read_sql(query_2021,connection)
df_2021.head()

Company_Brand  Founded HeadQuarter          Sector  \
0  Unbox Robotics   2019.0   Bangalore      AI startup   
1          upGrad   2015.0      Mumbai          EdTech   
2     Lead School   2012.0      Mumbai          EdTech   
3         Bizongo   2015.0      Mumbai  B2B E-commerce   
4        FypMoney   2021.0    Gurugram         FinTech   

                                        What_it_does  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   

                                            Founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   

                                        Investor        Amount         Stage  
0                    BEENEXT, Entrepreneur First    $1,200,000  Pre-series A  
1       Unilazer Ventures, IIFL Asset Management  $120,000,000          None  
2               GSV Ventures, Westbridge Capital   $30,000,000      Series D  
3                         CDC Group, IDG Capital   $51,000,000      Series C  
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000          Seed

In [5]:
# Load 2020 dataset from MS SQL Server
query_2020 = "SELECT * FROM dbo.LP1_startup_funding2020"
df_2020 = pd.read_sql(query_2020,connection)
df_2020.head()

Company_Brand  Founded HeadQuarter              Sector  \
0    Aqgromalin   2019.0     Chennai            AgriTech   
1      Krayonnz   2019.0   Bangalore              EdTech   
2  PadCare Labs   2018.0        Pune  Hygiene management   
3         NCOME   2020.0   New Delhi              Escrow   
4    Gramophone   2016.0      Indore            AgriTech   

                                        What_it_does  \
0                       Cultivating Ideas for Profit   
1  An academy-guardian-scholar centric ecosystem ...   
2   Converting bio-hazardous waste to harmless waste   
3                       Escrow-as-a-service platform   
4  Gramophone is an AgTech platform enabling acce...   

                                            Founders  \
0                    Prasanna Manogaran, Bharani C L   
1                   Saurabh Dixit, Gurudutt Upadhyay   
2                                    Ajinkya Dhariya   
3                                      Ritesh Tiwari   
4  Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   

                              Investor    Amount     Stage column10  
0                      Angel investors  200000.0      None     None  
1                      GSF Accelerator  100000.0  Pre-seed     None  
2                       Venture Center       NaN  Pre-seed     None  
3  Venture Catalysts, PointOne Capital  400000.0      None     None  
4  Siana Capital Management, Info Edge  340000.0      None     None

In [6]:
# load 2019 dataset
df_2019 = pd.read_csv("data/startup_funding2019.csv")
df_2019.head()

Company/Brand  Founded HeadQuarter           Sector  \
0  Bombay Shaving      NaN         NaN        Ecommerce   
1       Ruangguru   2014.0      Mumbai           Edtech   
2        Eduisfun      NaN      Mumbai           Edtech   
3        HomeLane   2014.0     Chennai  Interior design   
4        Nu Genes   2004.0   Telangana         AgriTech   

                                        What it does  \
0         Provides a range of male grooming products   
1  A learning platform that provides topic-based ...   
2            It aims to make learning fun via games.   
3              Provides interior designing solutions   
4  It is a seed company engaged in production, pr...   

                                Founders  \
0                     Shantanu Deshpande   
1  Adamas Belva Syah Devara, Iman Usman.   
2                          Jatin Solanki   
3           Srikanth Iyer, Rama Harinath   
4                 Narayana Reddy Punyala   

                                            Investor     Amount($)  \
0                               Sixth Sense Ventures    $6,300,000   
1                                   General Atlantic  $150,000,000   
2     Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3  Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4           Innovation in Food and Agriculture (IFA)    $6,000,000   

           Stage  
0            NaN  
1       Series C  
2  Fresh funding  
3       Series D  
4            NaN

In [7]:
# load 2018 dataset
df_2018 = pd.read_csv("./data/startup_funding2018.csv")
df_2018.head()

Company Name                                           Industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  Round/Series       Amount                          Location  \
0         Seed       250000       Bangalore, Karnataka, India   
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       About Company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

# Data Assesment
After the data was loaded from the various sources, the data was Visually and Programatically assessed on Quality Issues.

## Data Quality Issues
I identified several data quality issues while exploring the datasets
1. Incosistent and missing columns:Some datasets were identified to have incosistent columns structures,with missing columns.This applies to the 2018 dataset.
2. Missing values and duplicates: There exists some null values and duplicates within the indivdual datasets.
3. Incosistent Values and currencies in the amount column: The amount column contains incosistent values and different currency types. 
4. Incosistent Values in the Sector,Location.,Industry Column: The following columns contain a lot of inconsistent values which needs to be looked at.



# Data Cleaning on the 2018 Dataset

In [8]:
# load dataset
df_2018.head()

Company Name                                           Industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  Round/Series       Amount                          Location  \
0         Seed       250000       Bangalore, Karnataka, India   
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       About Company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

In [9]:
# rename column labels
df_2018 = df_2018.rename(columns=lambda x: x.lower().replace(" ","_"))
df_2018.head()

company_name                                           industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  round/series       amount                          location  \
0         Seed       250000       Bangalore, Karnataka, India   
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       about_company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

In [10]:
# rename  column
rename_columns= {
    "round/series":"stage",
    "industry":"sector",
    "about_company":"what_it_does"
}
df_2018 =df_2018.rename(columns=rename_columns) 

In [11]:
# confirm changes
df_2018.columns

Index(['company_name', 'sector', 'stage', 'amount', 'location',
       'what_it_does'],
      dtype='object')

In [12]:
# check the shape of the 2018 dataset
df_2018.shape

(526, 6)

In [13]:
# perform descriptive statistics on the dataset
df_2018.describe(include="all").T

count unique                                                top  \
company_name   526    525                                    TheCollegeFever   
sector         526    405                                                  —   
stage          526     21                                               Seed   
amount         526    198                                                  —   
location       526     50                        Bangalore, Karnataka, India   
what_it_does   526    524  TheCollegeFever is a hub for fun, fiesta and f...   

             freq  
company_name    2  
sector         30  
stage         280  
amount        148  
location      102  
what_it_does    2

In [14]:
# check the info about the dataset
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   company_name  526 non-null    object
 1   sector        526 non-null    object
 2   stage         526 non-null    object
 3   amount        526 non-null    object
 4   location      526 non-null    object
 5   what_it_does  526 non-null    object
dtypes: object(6)
memory usage: 24.8+ KB


## Observations on the 2018 dataset
- The dataset consists of 526 rows and 6 columns
- All the data types of the columns are of the object type
- The amount column must be a numeric value
- There exist some duplicates in the dataset

### Course of Action
- Convert the amount column data type to float
- Drop duplicated rows

In [15]:
# check for null values
df_2018.isna().sum()

company_name    0
sector          0
stage           0
amount          0
location        0
what_it_does    0
dtype: int64

In [16]:
# check for duplicated values
df_2018.duplicated().sum()

1

In [17]:
# drop the duolicate value
df_2018.drop_duplicates(inplace=True)


In [18]:
# verify if duplicate still exists
df_2018.duplicated().sum() 

0

## Cleaning of the Amount Column
Exchange Rate Between USD and Indian Rupee as of 2018

1USD = 68.4933INR

source: https://www.poundsterlinglive.com/history/USD-INR-2018

assumption: All the amount values without any currencies are assumed to be in USD

In [19]:
df_2018["amount"].unique()

array(['250000', '₹40,000,000', '₹65,000,000', '2000000', '—', '1600000',
       '₹16,000,000', '₹50,000,000', '₹100,000,000', '150000', '1100000',
       '₹500,000', '6000000', '650000', '₹35,000,000', '₹64,000,000',
       '₹20,000,000', '1000000', '5000000', '4000000', '₹30,000,000',
       '2800000', '1700000', '1300000', '₹5,000,000', '₹12,500,000',
       '₹15,000,000', '500000', '₹104,000,000', '₹45,000,000', '13400000',
       '₹25,000,000', '₹26,400,000', '₹8,000,000', '₹60,000', '9000000',
       '100000', '20000', '120000', '₹34,000,000', '₹342,000,000',
       '$143,145', '₹600,000,000', '$742,000,000', '₹1,000,000,000',
       '₹2,000,000,000', '$3,980,000', '$10,000', '₹100,000',
       '₹250,000,000', '$1,000,000,000', '$7,000,000', '$35,000,000',
       '₹550,000,000', '$28,500,000', '$2,000,000', '₹240,000,000',
       '₹120,000,000', '$2,400,000', '$30,000,000', '₹2,500,000,000',
       '$23,000,000', '$150,000', '$11,000,000', '₹44,000,000',
       '$3,240,000', '₹60

In [21]:
# create function to clean the amount column
exchange_rate = 68.4933
    
def clean_amount():
    # copy original amount columns
    amount_column = df_2018["amount"].copy().str.replace(",","") 
    # extract values in Rupees(₹)
    amount_in_rupee = amount_column[amount_column.str.startswith("₹")]
    # strip off the ₹ symbol
    amount_in_rupee = amount_in_rupee.str.lstrip("₹")
    # convert the amount in rupee to USD by the exchange rate of 68.4933
    amount_in_rupee = amount_in_rupee.apply(lambda x: float(x)/exchange_rate)
    # extract values in dollars($)
    amount_in_dollar = amount_column[amount_column.str.startswith("$")]
    # strip off the dollar symbol
    amount_in_dollar = amount_in_dollar.str.lstrip("$")
    # Replace the Unclean columns with the clean one
    amount_column.loc[amount_in_rupee.index] = amount_in_rupee
    amount_column.loc[amount_in_dollar.index] = amount_in_dollar
    
    # convert the clean column to numeric
    amount_column = pd.to_numeric(amount_column,errors="coerce")
    # update the amount column 
    df_2018["amount"] = amount_column




In [22]:
# call the clean_amount function
clean_amount()

In [23]:
# check info about the dataset after cleaning
df_2018.info()


<class 'pandas.core.frame.DataFrame'>
Index: 525 entries, 0 to 525
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   company_name  525 non-null    object 
 1   sector        525 non-null    object 
 2   stage         525 non-null    object 
 3   amount        377 non-null    float64
 4   location      525 non-null    object 
 5   what_it_does  525 non-null    object 
dtypes: float64(1), object(5)
memory usage: 44.9+ KB


In [24]:
# check for the null values present after cleaning the amount column
df_2018.isna().sum()

company_name      0
sector            0
stage             0
amount          148
location          0
what_it_does      0
dtype: int64

In [25]:
# perform a discriptive statistics on the new dataset
df_2018.describe(include="all").T

count unique                            top freq  \
company_name    525    525                TheCollegeFever    1   
sector          525    405                              —   30   
stage           525     21                           Seed  279   
amount        377.0    NaN                            NaN  NaN   
location        525     50    Bangalore, Karnataka, India  101   
what_it_does    525    524  Algorithmic trading platform.    2   

                         mean              std        min       25%  \
company_name              NaN              NaN        NaN       NaN   
sector                    NaN              NaN        NaN       NaN   
stage                     NaN              NaN        NaN       NaN   
amount        17616765.176312  77972595.460917  875.99809  500000.0   
location                  NaN              NaN        NaN       NaN   
what_it_does              NaN              NaN        NaN       NaN   

                    50%        75%           max  
company_name        NaN        NaN           NaN  
sector              NaN        NaN           NaN  
stage               NaN        NaN           NaN  
amount        1300000.0  5000000.0  1000000000.0  
location            NaN        NaN           NaN  
what_it_does        NaN        NaN           NaN

## Observations
- There is a vast difference between the mean and the median of the amount column which suggests outliers present in the dataset
- There exists some null values still present in the dataset 

## Course Of Actions
Fillna Values in the amount column:Impute the NaN values in the amount column with the median because of the outliers

In [26]:
# Function to fill NaN values in the amount column with the average based on stage of company
def impute_amount_column(df,filter_name,fill_value):
    unique_values = df[filter_name].unique()
    for val, avg_amount in zip(unique_values,df.groupby(filter_name)[fill_value].transform('median')):
        df.loc[df[filter_name] == val, fill_value] = df.loc[df[filter_name] == val, fill_value].fillna(avg_amount)
    return df
     


In [27]:
# call the impute_amount_column function
df_2018 = impute_amount_column(df_2018,"stage","amount")

In [28]:
df_2018.head()

company_name                                             sector  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

      stage        amount                          location  \
0      Seed  2.500000e+05       Bangalore, Karnataka, India   
1      Seed  5.839987e+05        Mumbai, Maharashtra, India   
2  Series A  9.489979e+05           Gurgaon, Haryana, India   
3     Angel  2.000000e+06       Noida, Uttar Pradesh, India   
4      Seed  5.839987e+05  Hyderabad, Andhra Pradesh, India   

                                        what_it_does  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

In [29]:
# check info about the cleaned dataset
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
Index: 525 entries, 0 to 525
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   company_name  525 non-null    object 
 1   sector        525 non-null    object 
 2   stage         525 non-null    object 
 3   amount        525 non-null    float64
 4   location      525 non-null    object 
 5   what_it_does  525 non-null    object 
dtypes: float64(1), object(5)
memory usage: 44.9+ KB


In [30]:
# Split the Industry Column
industry_split = df_2018["sector"].str.split(",", expand=True)
industry_split.head()

0                    1           2             3   \
0       Brand Marketing      Event Promotion   Marketing   Sponsorship   
1           Agriculture              Farming        None          None   
2                Credit   Financial Services     Lending   Marketplace   
3    Financial Services              FinTech        None          None   
4  E-Commerce Platforms               Retail        SaaS          None   

           4     5     6     7     8     9     10    11  
0   Ticketing  None  None  None  None  None  None  None  
1        None  None  None  None  None  None  None  None  
2        None  None  None  None  None  None  None  None  
3        None  None  None  None  None  None  None  None  
4        None  None  None  None  None  None  None  None

In [31]:
# concatenate industry_split with the original df_2018
df_2018 = pd.concat([df_2018,industry_split], ignore_index=False, axis=1)
df_2018.head()

company_name                                             sector  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

      stage        amount                          location  \
0      Seed  2.500000e+05       Bangalore, Karnataka, India   
1      Seed  5.839987e+05        Mumbai, Maharashtra, India   
2  Series A  9.489979e+05           Gurgaon, Haryana, India   
3     Angel  2.000000e+06       Noida, Uttar Pradesh, India   
4      Seed  5.839987e+05  Hyderabad, Andhra Pradesh, India   

                                        what_it_does                     0  \
0  TheCollegeFever is a hub for fun, fiesta and f...       Brand Marketing   
1  A startup which aggregates milk from dairy far...           Agriculture   
2          Leading Online Loans Marketplace in India                Credit   
3  PayMe India is an innovative FinTech organizat...    Financial Services   
4  Eunimart is a one stop solution for merchants ...  E-Commerce Platforms   

                     1           2             3           4     5     6  \
0      Event Promotion   Marketing   Sponsorship   Ticketing  None  None   
1              Farming        None          None        None  None  None   
2   Financial Services     Lending   Marketplace        None  None  None   
3              FinTech        None          None        None  None  None   
4               Retail        SaaS          None        None  None  None   

      7     8     9    10    11  
0  None  None  None  None  None  
1  None  None  None  None  None  
2  None  None  None  None  None  
3  None  None  None  None  None  
4  None  None  None  None  None

In [32]:
# drop the unwanted columns
df_2018.drop(columns=["sector",1,2,3,4,5,6,7,8,9,10,11],inplace=True)
df_2018.head()

company_name     stage        amount                          location  \
0  TheCollegeFever      Seed  2.500000e+05       Bangalore, Karnataka, India   
1  Happy Cow Dairy      Seed  5.839987e+05        Mumbai, Maharashtra, India   
2       MyLoanCare  Series A  9.489979e+05           Gurgaon, Haryana, India   
3      PayMe India     Angel  2.000000e+06       Noida, Uttar Pradesh, India   
4         Eunimart      Seed  5.839987e+05  Hyderabad, Andhra Pradesh, India   

                                        what_it_does                     0  
0  TheCollegeFever is a hub for fun, fiesta and f...       Brand Marketing  
1  A startup which aggregates milk from dairy far...           Agriculture  
2          Leading Online Loans Marketplace in India                Credit  
3  PayMe India is an innovative FinTech organizat...    Financial Services  
4  Eunimart is a one stop solution for merchants ...  E-Commerce Platforms

In [33]:
# Rename the column 0 to sector
df_2018 = df_2018.rename(columns={0:"sector"})
df_2018.head()

company_name     stage        amount                          location  \
0  TheCollegeFever      Seed  2.500000e+05       Bangalore, Karnataka, India   
1  Happy Cow Dairy      Seed  5.839987e+05        Mumbai, Maharashtra, India   
2       MyLoanCare  Series A  9.489979e+05           Gurgaon, Haryana, India   
3      PayMe India     Angel  2.000000e+06       Noida, Uttar Pradesh, India   
4         Eunimart      Seed  5.839987e+05  Hyderabad, Andhra Pradesh, India   

                                        what_it_does                sector  
0  TheCollegeFever is a hub for fun, fiesta and f...       Brand Marketing  
1  A startup which aggregates milk from dairy far...           Agriculture  
2          Leading Online Loans Marketplace in India                Credit  
3  PayMe India is an innovative FinTech organizat...    Financial Services  
4  Eunimart is a one stop solution for merchants ...  E-Commerce Platforms

In [34]:
# add founded column to the dataset
df_2018["founded"] = 2018
df_2018.head()

company_name     stage        amount                          location  \
0  TheCollegeFever      Seed  2.500000e+05       Bangalore, Karnataka, India   
1  Happy Cow Dairy      Seed  5.839987e+05        Mumbai, Maharashtra, India   
2       MyLoanCare  Series A  9.489979e+05           Gurgaon, Haryana, India   
3      PayMe India     Angel  2.000000e+06       Noida, Uttar Pradesh, India   
4         Eunimart      Seed  5.839987e+05  Hyderabad, Andhra Pradesh, India   

                                        what_it_does                sector  \
0  TheCollegeFever is a hub for fun, fiesta and f...       Brand Marketing   
1  A startup which aggregates milk from dairy far...           Agriculture   
2          Leading Online Loans Marketplace in India                Credit   
3  PayMe India is an innovative FinTech organizat...    Financial Services   
4  Eunimart is a one stop solution for merchants ...  E-Commerce Platforms   

   founded  
0     2018  
1     2018  
2     2018  
3     2018  
4     2018

In [35]:
# check info about the new dataset
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
Index: 525 entries, 0 to 525
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   company_name  525 non-null    object 
 1   stage         525 non-null    object 
 2   amount        525 non-null    float64
 3   location      525 non-null    object 
 4   what_it_does  525 non-null    object 
 5   sector        525 non-null    object 
 6   founded       525 non-null    int64  
dtypes: float64(1), int64(1), object(5)
memory usage: 49.0+ KB


In [36]:
# look for unique values in the sector column
df_2018["sector"].sort_values(ascending=True).unique()


array(['3D Printing', 'Accounting', 'Advertising', 'Aerospace', 'AgTech',
       'Agriculture', 'Air Transportation', 'Alternative Medicine',
       'Analytics', 'Android', 'Apps', 'Artificial Intelligence', 'Audio',
       'Automotive', 'Autonomous Vehicles', 'B2B', 'Banking',
       'Basketball', 'Battery', 'Beauty', 'Big Data', 'Biopharma',
       'Biotechnology', 'Blockchain', 'Brand Marketing', 'Broadcasting',
       'Business Development', 'Business Intelligence', 'Business Travel',
       'Career Planning', 'Catering', 'Child Care', 'Children',
       'Classifieds', 'Clean Energy', 'CleanTech', 'Cloud Computing',
       'Cloud Infrastructure', 'Collaboration', 'Commercial',
       'Commercial Real Estate', 'Communities', 'Computer', 'Consulting',
       'Consumer', 'Consumer Applications', 'Consumer Electronics',
       'Consumer Goods', 'Consumer Lending', 'Continuing Education',
       'Cooking', 'Cosmetics', 'Creative Agency', 'Credit',
       'Credit Cards', 'Crowdfunding', 

source: https://www.startupindia.gov.in/content/sih/en/government-schemes.html

 'Accounting', 'Advertising',
       'Agriculture', 'Air Transportation', 'Alternative Medicine',
       'Analytics', 'Android', 'Apps', 'Artificial Intelligence', 'Audio',
       'Automotive', 'Autonomous Vehicles', 'B2B', 'Banking',
       'Basketball', 'Battery', 'Beauty', 'Big Data', 'Biopharma',
       'Biotechnology', 'Blockchain', 'Brand Marketing', 'Broadcasting',
       'Business Development', 'Business Intelligence', 'Business Travel',
       'Career Planning', 'Catering', 'Child Care', 'Children',
       'Classifieds', 'Clean Energy', 'CleanTech', 'Cloud Computing',
       'Cloud Infrastructure', 'Collaboration', 'Commercial',
       'Commercial Real Estate', 'Communities', 'Computer', 'Consulting',
       'Consumer', 'Consumer Applications', 'Consumer Electronics',
       'Consumer Goods', 'Consumer Lending', 'Continuing Education',
       'Cooking', 'Cosmetics', 'Creative Agency', 'Credit',

{'3D Printing': 'Science & Technology',
 'Cleantech': 'Science & Technology',
 'Fintech': 'Science & Technology',
 'Smart Cities': 'Science & Technology',
 'Software': 'Science & Technology',
 'Search Engine': 'Science & Technology',
 'Social Media': 'Science & Technology',
 'Reading Appd Cities': 'Science & Technology',
 'Cloud Computing': 'Science & Technology',
 'Cloud Infrastructure': 'Science & Technology',
 'Blockchain': 'Science & Technology',
 'Analytics': 'Science & Technology',
 'Android': 'Science & Technology',
 'App': 'Science & Technology',
 'Artificial Intelligence': 'Science & Technology',
 'Big Data': 'Science & Technology',
 'Automative ': 'Science & Technology',
 'Autonomous Vehicle': 'Science & Technology'}

In [143]:
# Rename values in the sector column
df_2018["sector"] = df_2018["sector"].str.replace("Child Care","Children")
df_2018["sector"]= df_2018["sector"].str.replace("E-Commerce Platforms","E-Commerce")
df_2018["sector"] = df_2018["sector"].str.replace("Financial Services","Finance")
df_2018.head()

company_name     stage        amount                          location  \
0  TheCollegeFever      Seed  2.500000e+05       Bangalore, Karnataka, India   
1  Happy Cow Dairy      Seed  4.000000e+07        Mumbai, Maharashtra, India   
2       MyLoanCare  Series A  6.500000e+07           Gurgaon, Haryana, India   
3      PayMe India     Angel  2.000000e+06       Noida, Uttar Pradesh, India   
4         Eunimart      Seed  3.328377e+07  Hyderabad, Andhra Pradesh, India   

                                       about_company           sector  year  \
0  TheCollegeFever is a hub for fun, fiesta and f...  Brand Marketing  2018   
1  A startup which aggregates milk from dairy far...      Agriculture  2018   
2          Leading Online Loans Marketplace in India           Credit  2018   
3  PayMe India is an innovative FinTech organizat...          Finance  2018   
4  Eunimart is a one stop solution for merchants ...       E-Commerce  2018   

  founded  
0          
1          
2          
3          
4

In [144]:
# Check for unique values in the sector column again

df_2018["sector"].sort_values().unique()

array(['3D Printing', 'Accounting', 'Advertising', 'Aerospace', 'AgTech',
       'Agriculture', 'Air Transportation', 'Alternative Medicine',
       'Analytics', 'Android', 'Apps', 'Artificial Intelligence', 'Audio',
       'Automotive', 'Autonomous Vehicles', 'B2B', 'Banking',
       'Basketball', 'Battery', 'Beauty', 'Big Data', 'Biopharma',
       'Biotechnology', 'Blockchain', 'Brand Marketing', 'Broadcasting',
       'Business Development', 'Business Intelligence', 'Business Travel',
       'Career Planning', 'Catering', 'Children', 'Classifieds',
       'Clean Energy', 'CleanTech', 'Cloud Computing',
       'Cloud Infrastructure', 'Collaboration', 'Commercial',
       'Commercial Real Estate', 'Communities', 'Computer', 'Consulting',
       'Consumer', 'Consumer Applications', 'Consumer Electronics',
       'Consumer Goods', 'Consumer Lending', 'Continuing Education',
       'Cooking', 'Cosmetics', 'Creative Agency', 'Credit',
       'Credit Cards', 'Crowdfunding', 'Crowdsourcing

In [ ]:
sector_mapping = {
    '3D Printing': 'IT & Technology',
    'Accounting': 'Financial Services',
    'Advertising': 'Media & Entertainment',
    'Aerospace': 'Transportation & Logistics',
    'AgTech': 'Agriculture',
    'Agriculture': 'Agriculture',
    'Air Transportation': 'Transportation & Logistics',
    'Alternative Medicine': 'Healthcare',
    'Analytics': 'IT & Technology',
    'Android': 'IT & Technology',
    'Apps': 'IT & Technology',
    'Artificial Intelligence': 'IT & Technology',
    'Audio': 'Media & Entertainment',
    'Automotive': 'Transportation & Logistics',
    'Autonomous Vehicles': 'Transportation & Logistics',
    'B2B': 'Retail',
    'Banking': 'Financial Services',
    'Basketball': 'Sports',
    'Battery': 'IT & Technology',
    'Beauty': 'Consumer Goods',
    'Big Data': 'IT & Technology',
    'Biopharma': 'Healthcare',
    'Biotechnology': 'Healthcare',
    'Blockchain': 'IT & Technology',
    'Brand Marketing': 'Media & Entertainment',
    'Broadcasting': 'Media & Entertainment',
    'Business Development': 'Others',
    'Business Intelligence': 'IT & Technology',
    'Business Travel': 'Hospitality',
    'Career Planning': 'Education',
    'Catering': 'Hospitality',
    'Child Care': 'Healthcare',
    'Children': 'Consumer Goods',
    'Classifieds': 'Media & Entertainment',
    'Clean Energy': 'IT & Technology',
    'CleanTech': 'IT & Technology',
    'Cloud Computing': 'IT & Technology',
    'Cloud Infrastructure': 'IT & Technology',
    'Collaboration': 'IT & Technology',
    'Commercial': 'Others',
    'Commercial Real Estate': 'Others',
    'Communities': 'Others',
    'Computer': 'IT & Technology',
    'Consulting': 'Others',
    'Consumer': 'Consumer Goods',
    'Consumer Applications': 'Consumer Goods',
    'Consumer Electronics': 'Consumer Goods',
    'Consumer Goods': 'Consumer Goods',
    'Consumer Lending': 'Financial Services',
    'Continuing Education': 'Education',
    'Cooking': 'Consumer Goods',
    'Cosmetics': 'Consumer Goods',
    'Creative Agency': 'Media & Entertainment',
    'Credit': 'Financial Services',
    'Credit Cards': 'Financial Services',
    'Crowdfunding': 'Financial Services',
    'Crowdsourcing': 'IT & Technology',
    'Cryptocurrency': 'Financial Services',
    'Customer Service': 'Others',
    'Dating': 'Media & Entertainment',
    'Delivery': 'Transportation & Logistics',
    'Delivery Service': 'Transportation & Logistics',
    'Dental': 'Healthcare',
    'Dietary Supplements': 'Healthcare',
    'Digital Entertainment': 'Media & Entertainment',
    'Digital Marketing': 'Media & Entertainment',
    'Digital Media': 'Media & Entertainment',
    'E-Commerce': 'Retail',
    'E-Commerce Platforms': 'Retail',
    'E-Learning': 'Education',
    'EdTech': 'Education',
    'Education': 'Education',
    'Electric Vehicle': 'Transportation & Logistics',
    'Embedded Systems': 'IT & Technology',
    'Energy': 'IT & Technology',
    'Enterprise Resource Planning (ERP)': 'IT & Technology',
    'Enterprise Software': 'IT & Technology',
    'Environmental Consulting': 'IT & Technology',
    'Events': 'Hospitality',
    'Eyewear': 'Consumer Goods',
    'Facilities Support Services': 'Others',
    'Fantasy Sports': 'Sports',
    'Farming': 'Agriculture',
    'Fashion': 'Consumer Goods',
    'File Sharing': 'IT & Technology',
    'FinTech': 'Financial Services',
    'Finance': 'Financial Services',
    'Financial Services': 'Financial Services',
    'Fitness': 'Healthcare',
    'Food Delivery': 'Transportation & Logistics',
    'Gaming': 'Media & Entertainment',
    'Gift': 'Consumer Goods',
    'Grocery': 'Retail',
    'Health and Insurance': 'Healthcare',
    'Health and Wellness': 'Healthcare',
    'Health Care': 'Healthcare',
    'Health Diagnostics': 'Healthcare',
    'Health Services Industry': 'Healthcare',
    'Healthcare': 'Healthcare',
    'Home & Garden': 'Consumer Goods',
    'Home Decor': 'Consumer Goods',
    'Home Renovation': 'Consumer Goods',
    'Hospitals': 'Healthcare',
    'Hospitality': 'Hospitality',
    'Hotels': 'Hospitality',
    'Human Resources': 'Others',
    'Identity Management': 'IT & Technology',
    'Impact Investing': 'Financial Services',
    'Innovation Management': 'IT & Technology',
    'Insurance': 'Financial Services',
    'Intellectual Property': 'IT & Technology',
    'Internet': 'IT & Technology',
    'Investment Management': 'Financial Services',
    'IoT': 'IT & Technology',
    'iPad': 'IT & Technology',
    'iPhone': 'IT & Technology',
    'Journalism': 'Media & Entertainment',
    'Knowledge Management': 'IT & Technology',
    'Landscaping': 'Consumer Goods',
    'Language Learning': 'Education',
    'Lead Generation': 'IT & Technology',
    'Legal': 'Financial Services',
    'Leisure': 'Hospitality',
    'Lifestyle': 'Consumer Goods',
    'Linguistics': 'Education',
    'Local Businesses': 'Retail',
    'Logistics': 'Transportation & Logistics',
    'Market Research': 'IT & Technology',
    'Marketplace': 'Retail',
    'Media and Entertainment': 'Media & Entertainment',
    'Medical': 'Healthcare',
    'Medical Device': 'Healthcare',
    'Medical Professionals': 'Healthcare',
    'Medical Services': 'Healthcare',
    'Medical Supplies': 'Healthcare',
    'Medical Tourism': 'Healthcare',
    'Meeting Software': 'IT & Technology',
    'Mobile': 'IT & Technology',
    'Mobile Apps': 'IT & Technology',
    'Mobile Devices': 'IT & Technology',
    'Mobile Payments': 'Financial Services',
    'Music': 'Media & Entertainment',
    'Music Education': 'Education',
    'Music Services': 'Media & Entertainment',
    'Music Venues': 'Media & Entertainment',
    'Nanotechnology': 'IT & Technology',
    'Natural Language Processing': 'IT & Technology',
    'Network Security': 'IT & Technology',
    'Networking': 'IT & Technology',
    'Neuroscience': 'Healthcare',
    'News': 'Media & Entertainment',
    'Nutrition': 'Healthcare',
    'Online Auctions': 'Retail',
    'Online Portals': 'IT & Technology',
    'Online Reservations': 'Hospitality',
    'Online Rental': 'Retail',
    'Online Scheduling': 'IT & Technology',
    'Online Shopping': 'Retail',
    'Online Travel': 'Hospitality',
    'Open Source': 'IT & Technology',
        'Operating Systems': 'IT & Technology',
    'Optical Communication': 'IT & Technology',
    'Organic': 'Agriculture',
    'Outdoors': 'Consumer Goods',
    'Parenting': 'Education',
    'Payments': 'Financial Services',
    'Personal Branding': 'Others',
    'Personal Finance': 'Financial Services',
    'Personal Health': 'Healthcare',
    'Personalization': 'IT & Technology',
    'Pets': 'Consumer Goods',
    'Pharmaceuticals': 'Healthcare',
    'Photography': 'Media & Entertainment',
    'Physical Security': 'IT & Technology',
    'Plastics': 'Consumer Goods',
    'Podcasting': 'Media & Entertainment',
    'Point of Sale': 'Retail',
    'Politics': 'Media & Entertainment',
    'Predictive Analytics': 'IT & Technology',
    'Price Comparison': 'Retail',
    'Printing': 'IT & Technology',
    'Privacy': 'IT & Technology',
    'Private Social Networking': 'Media & Entertainment',
    'Procurement': 'IT & Technology',
    'Product Design': 'IT & Technology',
    'Product Development Services': 'IT & Technology',
    'Productivity Software': 'IT & Technology',
    'Professional Services': 'Others',
    'Project Management': 'IT & Technology',
    'Property Management': 'Real Estate',
    'Public Relations': 'Media & Entertainment',
    'Publishing': 'Media & Entertainment',
    'QR Codes': 'IT & Technology',
    'Real Estate': 'Real Estate',
    'Real Time': 'IT & Technology',
    'Recreation': 'Hospitality',
    'Recruiting': 'Others',
    'Renewable Energy': 'IT & Technology',
    'Rental': 'Retail',
    'Retail': 'Retail',
    'Search Engine': 'IT & Technology',
    'Smart Cities': 'IT & Technology',
    'Social Media': 'Media & Entertainment',
    'Software': 'IT & Technology',
    'Sports': 'Sports',
    'Tourism': 'Hospitality',
    'Trading Platform': 'Financial Services',
    'Training': 'Education',
    'Transportation': 'Transportation & Logistics',
    'Travel': 'Hospitality',
    'Veterinary': 'Healthcare',
    'Wealth Management': 'Financial Services',
    'Wedding': 'Hospitality',
    'Wellness': 'Healthcare',
    'eSports': 'Media & Entertainment',
    '—': 'Others'
}



In [145]:
# Split the location column
df_2018["headquarter","state","country"] = df_2018["location"].str.split(",",expand=True)
df_2018.head()

ValueError: Cannot set a DataFrame with multiple columns to the single column ('headquarter', 'state', 'country')

In [ ]:
# new sector dictionary
sector_dict = {{"Technology":
    ["Edtech","Agritech","Technology","Saas","AI&Tech","Fintech","HR tech","Food tech","Safety tech","Insurance technology","AI",
     "Food & tech","Automative tech","Legal tech","Cybersecurity","Healthtech","Infratech"]},
               {"Lifestyle":
    ["Social media","Cosmetics","Jewellery","Yoga & wellness","Accomodation","Fintech,'HR tech","Food tech","Safety tech","Insurance technology","AI",
     "Food & tech","Automative tech","Legal tech","Cybersecurity","Healthtech","Infratech"]}}